tests/31sync/03joined.pl

```diff
@@ -124,3 +124,65 @@
          Future->done(1)
       })
    };
+
+test "Newly joined room has correct timeline in incremental sync",
+   requires => [qw( first_api_client can_sync )],
+
+   check => sub {
+      my ( $http ) = @_;
+
+      my ( $user_a, $filter_id_a, $user_b, $filter_id_b, $room_id, $next_b );
+
+      my $filter = {
+         room => {
+            timeline => { types => [ "m.room.message" ], limit => 10 },
+            state    => { types => [] },
+         }
+      };
+
+      Future->needs_all(
```

Comment:
How much would you hate me if I asked you to phrase this as:

Future->need_all(
  matrix_register_user_with_filter( $http, $filter )->then( sub {
    ( $user_a, $filter_id_a ) = @_;
  }),
  matrix_register_user_with_filter( $http, $filter )->then( sub {
    ( $user_b, $filter_id_b ) = @_;
  }),
)->then(...

I don't really like that the output of the two calls are magically concatenated into a list for assorted reasons